templates/common/_base/files/ofport-request.yaml

```diff
@@ -0,0 +1,114 @@
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/pre-up.d/10-ofport-request.sh"
```

Comment:
With pre-up, has the interface been created in OVS already? Are we guaranteed that?

Reply:
According to the documentation: https://developer-old.gnome.org/NetworkManager/stable/NetworkManager.html
~~~
pre-up | The interface is connected to the network but is not         yet fully activated.  Scripts acting on this event must be placed or         symlinked into the /etc/NetworkManager/dispatcher.d/pre-up.d         directory, and NetworkManager will wait for script execution to complete before         indicating to applications that the interface is fully activated.
-- | --
~~~

So, `The interface is connected to the network`. There are several traces attached above with debug logging for NM where we can see that the interface is attached to OVS first, and pre-up is run only after that. I also tested on a dedicated virtual machine (RHEL 8.4) with the following setup with various port types (vlan, bond, standalone interfaces), and it works and the behavior is consistent:
~~~
[root@rhel84 ~]# ovs-vsctl show
0c29b398-2f10-497a-ba0f-ad834737cbd6
    Bridge br-ex
        Port eth2
            Interface eth2
                type: system
        Port vlan10
            Interface vlan10
                type: system
        Port br-ex
            Interface br-ex
                type: internal
        Port bond0
            Interface eth4
                type: system
            Interface eth5
                type: system
        Port patch-to-br-int
            Interface patch-to-br-int
                type: patch
                options: {peer=patch-to-br-ex}
        Port eth1
            Interface eth1
                type: system
        Port bond1
            Interface bond1
                type: system
    Bridge br-int
        Port patch-to-br-ex
            Interface patch-to-br-ex
                type: patch
                options: {peer=patch-to-br-int}
        Port br-int
            Interface br-int
                type: internal
    Bridge br-ex2
        Port eth3
            Interface eth3
                type: system
    ovs_version: "2.15.3"
[root@rhel84 ~]# nmcli conn
NAME                UUID                                  TYPE           DEVICE 
Wired connection 2  923b22f9-8cc8-32fa-aa0d-54cffb024215  ethernet       eth10  
Wired connection 3  2b1f9ef1-5fb3-35f5-9269-5ab7ef70f5c9  ethernet       eth8   
Wired connection 4  9432aa2c-d1aa-30a1-b852-970bc72b687d  ethernet       eth9   
Wired connection 1  a63944ec-3426-3ec1-b485-5b1603aadf3d  ethernet       eth0   
ovs-if-br-ex        74c03eca-f6a0-4409-b4e5-2698e0e0665a  ovs-interface  br-ex  
bond1               cd4c8622-8ccd-4a3e-b2a1-9b1d57667e80  bond           bond1  
bond1-eth6          23969750-49ef-4145-99c3-6af53b54f4a4  ethernet       eth6   
bond1-eth7          d20b59c8-99c8-4521-980f-9422b1682ab4  ethernet       eth7   
bond1-ovsport       1cc55059-da8f-4838-873c-6458e4c2682b  ovs-port       bond1  
br-ex               9e0055c1-23e7-473c-8444-015d4aa11dac  ovs-bridge     br-ex  
br-ex2              34109401-5f47-44ca-9f3b-5fc1b37422a5  ovs-bridge     br-ex2 
ovs-if-eth1         ec425a9c-9128-4d6e-a367-8cb0493f688f  ethernet       eth1   
ovs-if-eth2         c98f0470-7955-4195-bc0e-2174dc49085f  ethernet       eth2   
ovs-if-eth3         195a5964-2eff-4ead-8bda-0b071739e9d6  ethernet       eth3   
ovs-port-br-ex      0d18b6a6-2fe5-4c44-b705-d2fc4301aafc  ovs-port       br-ex  
ovs-port-br-ex2     8ba11cc7-a222-40f5-86a4-c569a85f81f4  ovs-port       br-ex2 
ovs-port-eth1       47871482-b875-4ba0-bb26-5e213135583f  ovs-port       eth1   
ovs-port-eth2       5b9bdfeb-7238-45dc-b7af-714b9a12ab11  ovs-port       eth2   
ovs-port-eth3       2f4f0995-7ead-4ffe-873c-e1f544ce1b7d  ovs-port       eth3   
ovs-slave-bond0     676b62a4-63b8-4c85-8cba-c1a091616e19  ovs-port       bond0  
ovs-slave-eth4      261b0374-1446-4f19-a2dd-5494b70212fe  ethernet       eth4   
ovs-slave-eth5      5d0fc00d-17e0-471a-b0a9-4bd851876d83  ethernet       eth5   
vlan10              4dfb2bb4-45ed-4fff-b746-6432874fde5a  vlan           vlan10 
vlan10-ovsport      1e7cf87e-a6c0-467c-ae88-70fb0e0d701b  ovs-port       vlan10 
ovs-if-br-ex2       4f70d1f0-48f7-42fa-b4bd-d7f69e4477de  ovs-interface  --   
~~~

All traces show that the interface is connected first (and gets a preassigned ID) and then the script kicks in.

Reply:
I threw in https://github.com/openshift/machine-config-operator/pull/2941/commits/9c63e55af5cca74291ae11a00d55ec17f31bb0a6#diff-bfb1096d30a23088db321d88004b44a807716f56cad84e920f3e5451f3d6f1e0R65   as a sanity check to exit gracefully if the interface does not exist.

Reply:
Conversation with Dan: https://coreos.slack.com/archives/CDCP2LA9L/p1645621075899249